alphastats/tl/differential_expression_analysis.py

```diff
@@ -0,0 +1,345 @@
+
+    @staticmethod
+    def _allowed_parameters() -> List[str]:
+        return [
+            DeaParameters.TEST_FUN,
+            DeaParameters.FDR_METHOD,
+            DeaParameters.GROUP1,
+            DeaParameters.GROUP2,
+            DeaParameters.GROUPING_COLUMN,
+            DeaParameters.METADATA,
+            PreprocessingStateKeys.LOG2_TRANSFORMED,
+        ]
+
+    def _extend_validation(self, parameters: dict):
+        """Validates the input and parameters for the t-test differential expression analysis.
+
+        This function checks for the required parameters for the t-test analysis, namely test_fun and fdr_method. The test_fun must be either scipy.stats.ttest_ind or scipy.stats.ttest_rel and the fdr_method must be one of 'bh' or 'by'.
+
+        Parameters:
+        parameters (dict): The parameters for the analysis.
+        """
+        super()._extend_validation(parameters)
+        if parameters["test_fun"] not in [
+            "independent",
+            "paired",
+        ]:
+            raise ValueError(
+                "test_fun must be either 'independent' for scipy.stats.ttest_ind or 'paired' for scipy.stats.ttest_rel."
+            )
+        if parameters["fdr_method"] not in ["fdr_bh", "bonferroni"]:
+            raise ValueError("fdr_method must be one of 'fdr_bh', 'bonferroni'.")
+
+    def _run_statistical_test(self, **kwargs) -> pd.DataFrame:
+        """Runs the t-test analysis and returns the result.
+        Wrapper to method with actual method parameters and implementation.
+
+        Returns:
+        pd.DataFrame: The result of the analysis.
+        """
+        group1, group2 = self._get_group_members(kwargs)
+        result = self._statistical_test_fun(
+            input_data=self.input_data,
+            group1=group1,
+            group2=group2,
+            is_log2_transformed=kwargs[PreprocessingStateKeys.LOG2_TRANSFORMED],
+            test_fun=kwargs[DeaParameters.TEST_FUN],
+            fdr_method=kwargs[DeaParameters.FDR_METHOD],
+        )
+        return result
+
+    @staticmethod
+    def _statistical_test_fun(
+        input_data: pd.DataFrame,
+        group1: list,
+        group2: list,
+        is_log2_transformed: bool,
+        test_fun: str,
+        fdr_method: str,
+    ) -> pd.DataFrame:
+        """Runs the t-test analysis and returns the result.
+
+        Parameters:
+        input_data (pd.DataFrame): The input data for the analysis.
+        group1 (list): The samples for group 1.
+        group2 (list): The samples for group 2.
+        is_log2_transformed (bool): Whether the data is log2 transformed.
+        test_fun (str): The test function to use, independent for scipy.stats.ttest_ind or paired for scipy.stats.ttest_rel.
+        fdr_method (str): The FDR method to use, 'fdr_bh' or 'bonferroni'.
+
+        Returns:
+        pd.DataFrame: The result of the analysis.
+        """
+        mat_transpose = input_data.loc[group1 + group2, :].transpose()
```

Comment:
just a thought: would it be possible to precompute indices for each group after the preprocessing? otherwise this could be slow, esp for larger dataframes